### PR TITLE
Add record type referencing example

### PR DESCRIPTION
**Pull request by pubudu91**

### CODE REVIEW

File: examples/record-type-reference/record_type_reference.bal [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import ballerina/io;
+
+type Person record {
+    string name;
+    int age = 25;
+    !...
+};
+
+type Employee record {
+    // Referencing the `Person` record
```
Comment (MaryamZi) [inline examples/record-type-reference/record_type_reference.bal]: Should this be a comment for `Employee` instead?

```ballerina
// Define a record `Employee` referencing the `Person` record.
type Employee record {
```

File: examples/record-type-reference/record_type_reference.bal [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    // Referencing the `Employee` record. Since `Employee` references `Person`,
+    // `Manager` will have the fields of `Person` and the additional fields
+    // in `Employee`.
+    *Employee;
+};
+
+public function main() {
+    // `Employee` has all the fields of `Person`.
+    Employee john = {name:"John Doe", designation:"Software Engineer"};
+    Employee jane = {name:"Jane Doe", designation:"UX Engineer"};
```
Comment (MaryamZi) [inline examples/record-type-reference/record_type_reference.bal]: need to fix formatting
```suggestion
    Employee jane = { name: "Jane Doe", designation: "UX Engineer" };
```

File: examples/record-type-reference/record_type_reference.description [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// Record type referencing provides a way convenient way to copy the fields defined in a record type descriptor
```
Comment (Shan1024) [inline examples/record-type-reference/record_type_reference.description]: a way convenient way -> a convenient way

File: examples/record-type-reference/record_type_reference.bal (hunk 1 above)
Comment (pubudu91) [reply examples/record-type-reference/record_type_reference.bal]: Hmm not really right? What type referencing does is copy the fields of the specified fields to the new record. And there can be multiple such references.

File: examples/record-type-reference/record_type_reference.bal [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    Employee[] team?;
+    // Referencing the `Employee` record. Since `Employee` references `Person`,
+    // `Manager` will have the fields of `Person` and the additional fields
+    // in `Employee`.
+    *Employee;
+};
+
+public function main() {
+    // `Employee` has all the fields of `Person`.
+    Employee john = {name:"John Doe", designation:"Software Engineer"};
```
Comment (Shan1024) [inline examples/record-type-reference/record_type_reference.bal]:
```suggestion
    Employee john = { name:" John Doe", designation:" Software Engineer" };
```

File: examples/record-type-reference/record_type_reference.bal (hunk 2 above)
Comment (Shan1024) [inline examples/record-type-reference/record_type_reference.bal]:
```suggestion
    Employee jane = { name:" Jane Doe", designation: "UX Engineer" };
```

File: examples/record-type-reference/record_type_reference.bal [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    Employee jane = {name:"Jane Doe", designation:"UX Engineer"};
+
+    // Type referencing copies the fields, including their properties
+    // (e.g., type, default value, optional status). As it can be seen
+    // by printing an `Employee` record, the optional field company
+    // is not included in the record.
+    io:println(john);
+    io:println(jane);
+
+    Manager mgr = {name:"Mark", age: 35, designation:"Engineering Manager"};
```
Comment (Shan1024) [inline examples/record-type-reference/record_type_reference.bal]:
```suggestion
    Manager mgr = { name: "Mark", age: 35, designation: "Engineering Manager" };
```

File: examples/record-type-reference/record_type_reference.bal (hunk 1 above)
Comment (Shan1024) [reply examples/record-type-reference/record_type_reference.bal]: I think it is better to add some kind of comment at the top as well.

**Pull request merged**